src/psij/__init__.py

```diff
@@ -41,26 +42,49 @@ def __init__(self, name: str, registration_method: Callable[[Type[T]], None]):
          _PluginType('launchers', Launcher.register_launcher)]
 
 
-def _load_plugins(path: str, type: _PluginType) -> None:
+def _load_plugins(root: str, path: str, _type: _PluginType) -> None:
     for mod in pkgutil.iter_modules(path=[path]):
         if not mod.ispkg and not mod.name[0] == '_':
-            im = importlib.import_module('psij.{}.{}'.format(type.name, mod.name))
-            var_name = '__PSI_J_{}__'.format(type.name.upper())
-            if hasattr(im, var_name):
-                classes = getattr(im, var_name)
-                logger.debug('Found module "{}" with classes {}'.format(mod.name, classes))
-                for cls in classes:
-                    logger.debug('Registering {}'.format(cls))
-                type.registration_method(cls)
-
-
-def _find_plugins(path: str, type: _PluginType, ix: int) -> None:
+            modname = 'psij.{}.{}'.format(_type.name, mod.name)
+            logger.debug('Attempting to load %s from %s', modname, path)
+            spec = mod.module_finder.find_spec(modname, None)
+            try:
+                if spec is None:
+                    raise Exception('Could not find module "%s"' % modname)
+                im = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(im)  # type: ignore
+                full_mod_path = spec.origin
+
+                var_name = '__PSI_J_{}__'.format(_type.name.upper())
+                if hasattr(im, var_name):
+                    classes = getattr(im, var_name)
+                    logger.debug('Found module "{}" with classes {}'.format(mod.name, classes))
+                    for cls in classes:
+                        if isinstance(cls, type) and issubclass(cls, JobExecutor):
```

Comment:
everything in the repo is ported to the new descriptor format so this code path and the launcher one below doesn't fire for anything in-repo at the moment, right?

Reply:
... unless you have an old version installed on the system, which I do, and which I'm keeping for now as a weird-environment-test-case.